geniza/annotations/views.py

```diff
@@ -0,0 +1,90 @@
+import json
+
+from django.http import HttpResponse, JsonResponse
+from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_http_methods
+from django.views.generic.base import View
+from django.views.generic.detail import SingleObjectMixin
+from django.views.generic.list import MultipleObjectMixin
+
+from geniza.annotations.models import Annotation
+
+
+class AnnotationResponse(JsonResponse):
+    content_type = 'application/ld+json; profile="http://www.w3.org/ns/anno.jsonld"'
+
+
+@method_decorator(csrf_exempt, name="dispatch")  # disable csrf for testing with curl
+class AnnotationList(View, MultipleObjectMixin):
+    model = Annotation
+    http_method_names = ["get", "post"]
+
+    paginate_by = None  # disable pagination for now
+
+    def get(self, request, *args, **kwargs):
+        # strictly speaking, annotations endpoint should return an annotation container,
+        # but that structure looks terrible to work with and we need a way to search,
+        # which is not defined in the w3c annotation protocol.
+
+        # implement something similar to SAS search by uri
+        annotations = self.get_queryset()
+        # if a target uri is specified, filter annotations
+        target_uri = self.request.GET.get("target_uri")
+        if target_uri:
+            annotations = annotations.filter(content__target__source__id=target_uri)
+
+        # return json response with list of annotations
+        # TODO: eventually we'll need to limit or paginate this
+        return AnnotationResponse(
+            {"items": [a.compile() for a in annotations]},
+        )
+
+    # todo check perms
+    def post(self, request, *args, **kwargs):
+        # parse request content as json
+        json_data = json.loads(request.body)
+        anno = Annotation()
+        anno.set_content(json_data)
+        anno.save()
+        resp = AnnotationResponse(anno.compile())
+        resp.status_code = 201  # created
+        # location header must include annotation's new uri
+        resp.location = anno.uri()
+
+        # TODO: should we create log entries to document activity?
```

Comment:
Yes!

Reply:
ok! was waffling on that, but agree it's probably a good idea. I should probably configure a minimal admin interface, then, too, so we have a way to investigate the content.